### PR TITLE
feat: add mmdit implementation based on sd3 paper

### PR DESCRIPTION
**Pull request by Fzilan**
# What does this PR do?

Adds # (feature)
Add mmdit implementation for image based on [sd3 research paper](https://arxiv.org/pdf/2403.03206.pdf).

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@geniuspatrick @SamitHuang @CaitinZhao

### CODE REVIEW

File: mindone/models/mmdit.py [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+        imgs = x.reshape((x.shape[0], c, h * p, h * p))
+        return imgs
+
+    def construct(
+        self,
+        x: Tensor,
+        t: Tensor,
+        clip_emb: Tensor,
+        clip_pooled_emb: Tensor,
+        t5_emb: Tensor = None,
```
Comment (Fzilan) [inline mindone/models/mmdit.py:395]: + [optional]

File: mindone/models/mmdit.py [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+    """A diffusion model with a Transformer backbone.
+    Args:
+        input_size (int, default=32): The size of the input latent.
+        patch_size (int, default=2): The size of each patch in the input latent. The input latent is divided into patches of patch_size x patch_size.
+        in_channels (int, default=4): The number of input channels in the input latent.
+        hidden_size (int, default=1152): The hidden size of the Transformer model.
+        depth (int, default=28): The number of blocks in this Transformer.
+        num_heads (int, default=16): The number of attention heads.
+        mlp_ratio (float, default=4.0): The expansion ratio for the hidden dimension in the MLP of the Transformer.
+        class_dropout_prob (float, default=0.1): The dropout probability for the class labels in the label embedder.
```
Comment (Fzilan) [inline mindone/models/mmdit.py:255]: unuse, delete

File: mindone/models/mmdit.py [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+
+        for block in self.blocks:
+            x, c = block(x, c, t, spatial_freq=self.spatial_freq)  # (N, T, D)
+
+        x = self.final_layer(x, t)  # (N, T, patch_size ** 2 * out_channels)
+        x = self.unpatchify(x)  # (N, out_channels, H, W)
+        return x
+
+    @ms.jit
+    def construct_with_cfg(
```
[the remark this answers is not in the document]
Comment (Fzilan) [reply mindone/models/mmdit.py]: done

File: mindone/models/mmdit.py (hunk 1 above)
Comment (Fzilan) [reply mindone/models/mmdit.py]: done

File: mindone/models/mmdit.py (hunk 2 above)
Comment (Fzilan) [reply mindone/models/mmdit.py]: done

File: mindone/models/mmdit.py [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+        return imgs
+
+    def construct(
+        self,
+        x: Tensor,
+        t: Tensor,
+        clip_emb: Tensor,
+        clip_pooled_emb: Tensor,
+        t5_emb: Tensor = None,
+        **kwargs,
```
[the remark this answers is not in the document]
Comment (Fzilan) [reply mindone/models/mmdit.py]: done

**Review by geniuspatrick — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**